src/PrestaShopBundle/Controller/Admin/Sell/Order/CartController.php

```diff
@@ -620,4 +623,32 @@ private function getErrorCode(Exception $e): int
 
         return Response::HTTP_INTERNAL_SERVER_ERROR;
     }
+
+    /**
+     * This method will be removed in the next patch version. We rely on Cart ObjectModel to simplify the code
+     *
+     * @param int $cartId
+     * @param int $productId
+     * @param int|null $combinationId
+     *
+     * @return int
+     */
+    private function getProductGiftedQuantity(int $cartId, int $productId, ?int $combinationId): int
+    {
+        $giftedQuantity = 0;
+        $cart = new \Cart($cartId);
+        $giftCartRules = $cart->getCartRules(\CartRule::FILTER_ACTION_GIFT, false);
```

Comment:
Move to `use` section

Reply:
same explanation as the question by @matks 
It's temporary code, didn't want to pollute the *use* section